src/commands/generate/ckl_metadata.ts

```diff
@@ -1,10 +1,51 @@
 import {Command, Flags} from '@oclif/core'
 import fs from 'fs'
 import promptSync from 'prompt-sync'
-import _ from 'lodash'
+import {Assettype, ChecklistMetadata, Role, Techarea, validateChecklistMetadata} from '@mitre/hdf-converters'
+import path from 'path'
 
 const prompt = promptSync()
 
+// Ensures that no empty strings are passed into the metadata
+// by returning undefined instead
+function enforceNonEmptyString(ask: string) : string | undefined {
+  const response = prompt({ask})
+  if (response)
+    return response
+  return undefined
+}
+
+function enforceInteger(ask: string): number {
+  let response = prompt({ask})
+  let intRep: number
+  while (true) {
+    intRep = Number.parseInt(response, 10)
+    const floatRep = Number.parseFloat(response)
+    if (intRep === floatRep && intRep >= 0 && !Number.isNaN(intRep))
+      break
+    console.log(`${response} is not a valid non-negative integer. Please try again`)
+    response = prompt({ask})
+  }
+
+  return intRep
+}
+
+function enforceEnum(ask: string, options: string[]): string | undefined {
+  // format prompt to show valid options (removes empty string options)
+  ask = `${ask} (${options.filter(Boolean).join('/')}) `
+  let response = prompt({ask})
+  while (true) {
+    if (options.includes(response))
+      break
+    if (!response)
+      return
+    console.log(`${response} is not a valid option. Please try again.`)
```

Comment:
add a parenthetical here about how the user needs to make sure capitalization and everything matches as well

OR

turn the enum and the user supplied value to lowercase, compare equality, and then put the enum value in there if it worked.  iirc there's actually a vulnerability if you put in the user supplied value since doing the lowercasing might cause disallowed uppercase characters through.

Reply:
Done